pkg/webhook/server.go

```diff
@@ -82,6 +86,74 @@ func (s *Server) setDefaults() {
 	}
 }
 
+// For registers defaulting and validation webhooks for the provided types that implement either
+// webhooktuil.validator or webhookutil.defaulter.
+func (s *Server) EnableDefaultsFor(scheme *runtime.Scheme, objects ...runtime.Object) error {
+	// Create a webhook for each type
+	for _, object := range objects {
+		wh := newMutatingWebhookFor(object)
+		if wh == nil {
+			continue
+		}
+
+		gvks, _, err := scheme.ObjectKinds(object)
+		if err != nil {
+			return err
+		}
+		if len(gvks) != 1 {
+			return fmt.Errorf("expected only GVK returned by scheme.ObjectKinds")
+		}
+		fmt.Println("/mutate-" + generatePath(gvks[0]))
+		s.Register("/mutate-"+generatePath(gvks[0]), wh)
+	}
+	return nil
+}
+
+// For registers defaulting and validation webhooks for the provided types that implement either
+// webhooktuil.validator or webhookutil.defaulter.
+func (s *Server) EnableValidationFor(scheme *runtime.Scheme, objects ...runtime.Object) error {
+	// Create a webhook for each type
+	for _, object := range objects {
+		wh := newValidatingWebhookFor(object)
+		if wh == nil {
+			continue
+		}
+
+		gvks, _, err := scheme.ObjectKinds(object)
+		if err != nil {
+			return err
+		}
+		if len(gvks) != 1 {
+			return fmt.Errorf("expected only GVK returned by scheme.ObjectKinds")
+		}
+		fmt.Println("/validate-" + generatePath(gvks[0]))
+		s.Register("/validate-"+generatePath(gvks[0]), wh)
+	}
+	return nil
+}
```

Comment:
I am sort of wondering if we should enable validation/defaulting by default when webhook server starts ? (This will save people one step if they want to enable validation/defaulting). And I am thinking of doing the same thing for conversion.